frontend/src/components/TaskCreator/DatePicker.tsx

```diff
@@ -356,13 +356,14 @@ export default function DatePicker(props: Props): ReactElement {
       endDate = internalDate.repeatEnd.date;
     } else {
       // TODO once database support exists, replace this with number of occurrances
-      endDate = new Date(
-        new Date().getTime() + 1000 * 60 * 60 * 24 * 7 * internalDate.repeatEnd.weeks,
-      );
+      endDate = new Date(new Date(
+        new Date().getTime() + 1000 * 60 * 60 * 24 * 7 * internalDate.repeatEnd.weeks
+        - 1000 * 60 * 60 * 24,
+      ).setHours(23, 59, 59, 999));
     }
 
     const repData: RepeatMetaData = {
-      startDate: new Date(),
+      startDate: new Date(new Date().setHours(0, 0, 0, 0)),
```

Comment:
We started to move away from using `new Date()` everywhere in https://github.com/cornell-dti/samwise/pull/328?
Could you move it to the react hook instead?